src/controllers/expensesController.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+const { validateExpenseBody, filterExpenses } = require('../utils/helpers');
+const { isUser } = require('./userController');
+
+let idTracker = 1;
+
+let expenses = [];
+
+const resetExpenses = () => {
+  expenses = [];
+};
+
+const getAllExpenses = (req, res) => {
+  const filteredExpenses = filterExpenses(expenses, req.query);
+
+  res.json(filteredExpenses);
+};
+
+const createNewExpense = (req, res) => {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+
+  if (!validateExpenseBody(req.body) || !isUser(userId)) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newExpense = {
+    id: idTracker,
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  };
+
+  idTracker++;
+
+  expenses.push(newExpense);
+  res.status(201).json(newExpense);
+};
+
+const getExpense = (req, res) => {
+  const { id } = req.params;
+
+  const foundExpense = expenses.find((user) => user.id === +id);
+
+  if (!foundExpense) {
+    res.sendStatus(404);
+  } else {
+    res.json(foundExpense);
+  }
+};
+
+const deleteExpense = (req, res) => {
+  const { id } = req.params;
+
+  const expenseIndex = expenses.findIndex((user) => user.id === +id);
```

Comment:
The variable name in the findIndex method should be 'expense' not 'user' to avoid confusion